src/main/java/duke/TaskManager.java

```diff
@@ -0,0 +1,96 @@
+package duke;
+
+import duke.tasktype.Deadline;
+import duke.tasktype.Event;
+import duke.tasktype.ToDo;
+
+import java.util.ArrayList;
+
+public class TaskManager {
+    protected ArrayList<Task> tasks;
+    protected int noOfTasks;
+
+    public TaskManager() {
+        this.tasks = new ArrayList<>();
+        this.noOfTasks = 0;
+    }
+
+    private Task identifyTaskType(String line) throws StringIndexOutOfBoundsException {
+        if (!(line.contains(" "))) {
+            throw new StringIndexOutOfBoundsException();
+        }
+        if (line.startsWith("todo")) {
+            String description = line.substring(line.indexOf(" ") + 1);
+            return new ToDo(description);
+        } else {
+            if (!(line.contains("/")) || (line.indexOf("/") + 1 == line.length())) {
+                throw new StringIndexOutOfBoundsException();
+            }
+            String timeline = line.substring(line.indexOf("/") + 1);
+            String description = line.substring(line.indexOf(" ") + 1, line.indexOf("/"));
+            if (line.startsWith("deadline")) {
+                return new Deadline(description, timeline);
+            } else {
+                return new Event(description, timeline);
+            }
+        }
+    }
+
+    public void addTask(String line) throws StringIndexOutOfBoundsException {
+        Task addedTask = identifyTaskType(line);
+        tasks.add(addedTask);
+        System.out.println("Noted! I have added this task:");
+        System.out.println(tasks.get(noOfTasks));
+        System.out.println("Now you have " + (noOfTasks + 1) + " tasks in your list.");
```

Comment:
perhaps practice SLAP concept here? e.g one method to print whether the current action was successful or not..